### PR TITLE
add trapping rain water.

### PR DESCRIPTION
**Pull request by alexpantyukhin**
#### Description of Change

Add a guide for new LeetCode solutions.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: leetcode/src/42.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
+int max(int a, int b){
```
Comment (Panquesito7) [inline leetcode/src/42.c:1]: Could you please add a small description of the algorithm?

Comment (alexpantyukhin) [reply leetcode/src/42.c:1]: I added the description

**Comment by alexpantyukhin** [comment]
@Panquesito7 The conflict has been fxied.

File: leetcode/src/42.c [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    int leftMax = 0;
+    int result = 0;
+    for (int i = 0; i < heightSize; i++){
+        leftMax = max(leftMax, height[i]);
+        result += max(0, min(leftMax, rightMaxStack[i]) - height[i]);
+    }
+    
+    free(rightMaxStack);
+    return result;
+}
```
Comment (Panquesito7) [inline leetcode/src/42.c:27]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
Looks great. Thanks! 🚀

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**